src/main/scala/org/constellation/lb/Manager.scala

```diff
@@ -231,3 +242,18 @@ class Manager(init: NonEmptyList[Addr], config: LoadbalancerConfig)(
       )
   }
 }
+
+object Manager {
+
+  def selectSameSessionNodes(knownSession: Option[Long], currentHosts: NonEmptyMap[Addr, Option[List[Info]]]): (SortedMap[Addr, Option[List[Info]]], SortedMap[Addr, Option[List[Info]]])  = {
+    val (sameSessionNodes, otherSessionNodes) = currentHosts.toSortedMap.partition {
+      case (_, v) =>
+        knownSession.flatMap(session => v.map(_.forall(_.clusterSession == session))).getOrElse(true)
```

Comment:
true or false, it doesn't matter, it means knownSession is None, which is actually a pathological case (it should never happen) . Since there's no session to discriminate, all values of the partition will either fall on the first collection (if we default to true) or in the second collection (if we default to false)

Reply:
Actually, it matters, it needs to be true: if all the nodes go down, we won't find any session, but instead of filter out everything, it just keeps the collection unchanged (if we set to false, we'll remove all the addresses and the load balancer won't be able to recover)

Reply:
Hm, I got it. then, maybe it's not worth to have `Option[Long]` but just `Long` if we should always know the session. otherwise, it's a fetch or unexpected error and we should not proceed

Reply:
Hmmm. If all nodes go down we probably should go back to source nodes again. Otherwise load balancer may stuck in a fork

Reply:
yeah, that's actually an argument for always consider seed nodes only

Reply:
I've modified the logic, if the session cannot be determined (seed nodes are down) no discovery is performed, i,e, no new nodes are added.